keepassxc: fix build on Darwin, fix test timeout

eliandoran opened the pull request:
###### Description of changes

### 1. Fix Darwin build

The build would fail on `x86_64-darwin` with the following error:

```
kCFURLBookmarkCreationWithSecurityScope API_AVAILABLE(macos(10.7), macCatalyst(13.0))  API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 11 ), // Mac OS X 10.7.3 and later, include information in the bookmark data which allows the same sandboxed process to access the resource after being relaunched
```

Applied the fixes according to the discussion in https://github.com/NixOS/nixpkgs/pull/210062. Used as inspiration the fix for `kotatogram-desktop` by @ilya-fedin in https://github.com/NixOS/nixpkgs/commit/77d99717a6ca9e1686597d383d8954244a912dfd .

### 2. Fix test timeout

The test was failing because it was timing out. Turns out it was waiting
for `foo.kdbx`, which couldn't be "seen" even if it actually existed
(probably some contrast issues with the theme and OCR couldn't find it).

Fixed it by delegating the check to the next screen, where the full path
to the file is displayed in a bigger size. The test seems to pass.

see #192247.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

eliandoran committed:
keepassxc: fix build on darwin

ck3d commented:
Result of `nixpkgs-review pr 217000` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

turion commented:
Any idea why the linux integration tests fail now?

rapenne-s commented:
> Any idea why the linux integration tests fail now?

that's not new https://github.com/NixOS/nixpkgs/issues/192247

turion commented:
Unfortunately the darwin tests don't even evaluate: The following builds were skipped because they don't evaluate on aarch64-darwin: keepassxc.passthru.tests

Ofborg reports a false success here.

eliandoran committed:
keepassxc: fix test timeout

The test was failing because it was timing out. Turns out it was waiting
for `foo.kdbx`, which couldn't be "seen" even if it actually existed
(probably some contrast issues with the theme and OCR couldn't find it).
Fixed it by delegating the check to the next screen, where the full path
to the file is displayed in a bigger size. The test seems to pass.

eliandoran commented:
@turion , added a potential fix for the Linux integration tests.

You've mentioned darwin tests, but I don't think the current integration tests for `keepassxc` has darwin support. It's using the x11. Would it be relevant under darwin?

turion commented:
@ofborg test keepassxc

turion commented:
> You've mentioned darwin tests, but I don't think the current integration test for keepassxc has darwin support. It's using the x11.

You're right, I misread the PR description. Thanks a lot, it seems the tests are now fixed! :tada:

turion commented on pkgs/applications/misc/keepassx/community.nix:
```diff
@@ -110,7 +112,7 @@ stdenv.mkDerivation rec {
     readline
     zlib
   ]
-  ++ lib.optional (stdenv.isDarwin && withKeePassTouchID) darwin.apple_sdk.frameworks.LocalAuthentication
```
Are we still using the `darwin` input to this package after this change?

eliandoran replied:
@turion , good point. Let me check.

eliandoran replied:
@turion , removed the `darwin` input since it was no longer needed.

turion commented on pkgs/top-level/all-packages.nix:
```diff
@@ -29266,7 +29266,10 @@ with pkgs;
   keepassx = callPackage ../applications/misc/keepassx { };
   keepassx2 = callPackage ../applications/misc/keepassx/2.0.nix { };
-  keepassxc = libsForQt5.callPackage ../applications/misc/keepassx/community.nix { };
+  keepassxc = libsForQt5.callPackage ../applications/misc/keepassx/community.nix {
+    inherit (darwin.apple_sdk_11_0.frameworks) LocalAuthentication;
+    stdenv = if stdenv.isDarwin then darwin.apple_sdk_11_0.stdenv else stdenv;
+  };
```
Is there an advantage to doing this here instead of in `pkgs/applications/misc/keepassx/community.nix`?

eliandoran replied:
@turion , I thought about that too. The only advantage I could see to keeping it here is the consistency with other packages.

I saw the same pattern with 37 other packages, in `all-packages.nix`, by searching for `apple_sdk_11_0.frameworks`.

turion replied:
Maybe there is a difference between overriding `libsForQt5.callPackage` and `callPackage`, but if there is I don't know it.

Ok, then let's keep it consistent and hope that whoever will update the SDK eventually will also remember to search for that code :sweat_smile:

turion commented on pkgs/top-level/all-packages.nix:
```diff
@@ -29266,7 +29266,10 @@ with pkgs;
   keepassx2 = callPackage ../applications/misc/keepassx/2.0.nix { };
-  keepassxc = libsForQt5.callPackage ../applications/misc/keepassx/community.nix { };
+  keepassxc = libsForQt5.callPackage ../applications/misc/keepassx/community.nix {
+    inherit (darwin.apple_sdk_11_0.frameworks) LocalAuthentication;
```
This will probably go out of sync once SDK 11.0 is outdated, but I don't know any other solution

eliandoran replied:
@turion , agreed. Probably a repository-wide change will be made when a new SDK will come along, since multiple packages are using the same pattern. I'm not aware of any other solution than using SDK 11 to avoid the build errors.

turion replied:
Fair enough.

turion commented:
Thanks again!
